lsコマンドの課題を提出いたします

ham-cap opened the pull request:
lsコマンドを作成いたしましたのでご確認願います。

対応オプション：`-a`、`-l`、`-r`
歓迎要件への対応：なし

よろしくお願いいたします。

ham-cap committed:
LsFileクラスとdefaultメソッドを仮実装

カレントディレクトリのファイル情報をLsFileクラスにまとめ、そのインスタンスで配列を用意。
オプションを指定せずに実行した場合の処理をdefaultメソッドとして定義した。

ham-cap committed:
-a、-rオプションを実装

ham-cap committed:
-lオプションを実装（途中）

LsFileクラス内のftype, modeメソッドを修正。user_name, group_nameメソッドを追加。

ham-cap committed:
-lオプションのタイムスタンプ部分を実装

タイムスタンプの部分を実装した。日と月は一桁の場合に左側に空白文字を挿入した。

ham-cap committed:
-lオプションの出力部分を実装

-lオプションを使用した場合の出力内容を実装。
オプション有無の判定は未実装。

ham-cap committed:
-lオプションを実装（完了）

-lオプションに必要なすべての項目表示を実装。
プログラムの実行部分も実装。

ham-cap committed:
rubocopの指摘を反映

case文による分岐をハッシュを用いて値を取り出す方式へ変更
長すぎる変数名を修正
その他、インデント等の軽微な修正

ham-cap committed:
デフォルトの表示（-lオプション未使用時）を修正

試しに4列表示にしてみたコードがそのまま残っていたため課題の必須要件である3列表示に修正。

yuuu commented on 05.ls/ls.rb:
```diff
@@ -0,0 +1,112 @@
+    @owner_permission, @group_permission, @others_permission =
+      @file_info.mode.to_s(8).slice(/\d\d\d$/).split(//)
+
+    @permission_list = { '0' => '---', '1' => '--x', '2' => '-w-', '3' => '-wx', '4' => 'r--', '5' => 'r-x', '6' => 'rw-', '7' => 'rwx' }
```
この変数はインスタンス変数にしなくても差し支えない(`@` を外しても動く)と思われます。

yuuu commented on 05.ls/ls.rb:
```diff
@@ -0,0 +1,112 @@
+
+  def display_columns
+    names = []
+    @files.each do |file|
```
Arrayから別のArrayを作る処理なので、`Array#map` を活用しましょう。

yuuu commented on 05.ls/ls.rb:
```diff
@@ -0,0 +1,112 @@
+
+  def display_in_details
+    @total = 0
+    @files.each do |file|
```
合計値を求める処理は `Array#sum` を使うとスッキリ書けます。

yuuu commented on 05.ls/ls.rb:
```diff
@@ -0,0 +1,112 @@
+  end
+end
+
+class LsArray < Array
```
`Array` クラスのメソッドや属性を利用しているわけではなさそうなので、継承しなくても動きそうです。不要であれば継承部分は削除してください。

yuuu commented on 05.ls/ls.rb:
```diff
@@ -0,0 +1,112 @@
+  end
+
+  def display_in_details
+    @total = 0
```
この変数もインスタンス変数としなくても良さそうですね。

ham-cap committed:
レビューの指摘事項5点を修正

レビューにて指摘のあった以下の事項を修正した。
・他のメソッドで使用されないインスタンス変数をローカル変数へ変更(×2)
・display_columnメソッド内の処理をeachを使用したものからmapを使用する方法へ変更
・display_in_detailsメソッド内の処理をeachを使用したものからmapとsumを使用する方法へ変更
・LsArrayクラスはArrayクラスを継承する必要がないことが確認できたため継承部分を削除

ham-cap commented:
ご指摘いただいた部分を修正いたしました。
rubocopも再度通しています。
ご確認の程よろしくお願いいたします。
